packages/react-utils/src/useSticky.ts

```diff
@@ -132,11 +132,17 @@ export const useSticky = (
       if (!when || !side) return
 
       if (isI1Intersecting && !isI2Intersecting) {
-        const parent = container?.current ?? document.body
+        const parent = container?.current
         const element = ref.current
-        if (!(parent instanceof HTMLElement) || !(element instanceof HTMLElement)) return
+        if (!(element instanceof HTMLElement)) return
+
+        const pRect = parent?.getBoundingClientRect() ?? {
+          top: 0,
+          left: 0,
+          bottom: window.innerHeight,
+          right: window.innerWidth,
+        }
```

Comment:
suggestion: You could do `new DOMRect(x, y, width, height)` instead of object for stricter typings :) https://developer.mozilla.org/en-US/docs/Web/API/DOMRect/DOMRect#syntax